src/backend/distributed/planner/distributed_planner.c

```diff
@@ -195,11 +195,20 @@ distributed_planner(Query *parse, int cursorOptions, ParamListInfo boundParams)
 		}
 		else
 		{
-			DistributedPlan *delegatePlan = TryToDelegateFunctionCall(parse);
+			bool hasParam = false;
+			DistributedPlan *delegatePlan = TryToDelegateFunctionCall(parse, &hasParam);
 			if (delegatePlan != NULL)
 			{
 				result = FinalizePlan(result, delegatePlan);
 			}
+			else if (hasParam)
+			{
+				/*
+				 * As in CreateDistributedPlannedStmt, try dissuade planner when planning
+				 * potentially failed due to unresolved prepared statement parameters.
+				 */
+				result->planTree->total_cost = FLT_MAX / 100000000;
```

Comment:
Wouldn't this apply to all regular  postgres plans as well?

Reply:
Ah, no, the check is done quite late. Maybe isDistributedFunctionCallWithParam?

Reply:
We only set `hasParam` to true if we get rather deep into `TryToDelegateFunctionCall`